joomla/joomla-cms
# [4.0] Alerts custom element redo

Redo of Pull Request #33234 .

## This one should be treated as Release Blocker 
(The alert code wasn't production-grade...)

### Summary of Changes
The Alert Custom Element had some known issues and this PR is pulling a version that fixes all of them https://github.com/joomla-projects/custom-elements/pull/179 


### Testing Instructions
Go to front end and backend and check that alerts are functioning as before (there are no styling changes, although the source is using CSS Vars for almost all parameters)

You can fire an alert using the browser's console with these snippets: 

```js

// Danger
Joomla.renderMessages({danger: ['Something happen']})
// Warning
Joomla.renderMessages({warning: ['Something happen']})
// Success
Joomla.renderMessages({success: ['Something happen']})
// Info
Joomla.renderMessages({info: ['Something happen']})

```

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request


![Screenshot 2021-07-14 at 12 38 57](https://user-images.githubusercontent.com/3889375/125609783-da8f5415-a39b-4c95-b10f-8cdbc1fc7b7d.png)
![Screenshot 2021-07-14 at 12 39 12](https://user-images.githubusercontent.com/3889375/125609786-17236edc-6ed7-43ee-aae4-a71d0bac0a38.png)

### Documentation Changes Required

The way that messages are rendered is exactly the same (only one internal change, not exposed to the API), so there is not anything to document.


@wilsonge DO NOT MERGE THIS WITHOUT MERGING THE OTHER PR CREATE A RELEASE AND CHANGE THE PACKAGE.JSON TO USE A VERSION!!!!
